### PR TITLE
Several minor improvements

### PR DESCRIPTION
**Pull request by kostaspap**
## Background

Several style improvements

## Changes

- Several style improvements 

## Testing

- `mage test:ci`
- `mage deploy`

### CODE REVIEW

File: internal/core/source_api/api/check_integration.go [hunk 1]
```diff
@@ -49,37 +49,47 @@ var (
...
+}
 
-	return out, nil
+func checkAwsScanIntegration(input *models.CheckIntegrationInput) *models.SourceIntegrationHealth {
+	out := &models.SourceIntegrationHealth{
+		AWSAccountID:    aws.StringValue(input.AWSAccountID),
+		IntegrationType: aws.StringValue(input.IntegrationType),
+	}
+	_, out.AuditRoleStatus = getCredentialsWithStatus(fmt.Sprintf(auditRoleFormat,
+		*input.AWSAccountID, *awsSession.Config.Region))
```
Comment (nhakmiller) [inline internal/core/source_api/api/check_integration.go:68]: `input.AWSAccontID` could be nil now right? Won't this panic?

File: internal/core/source_api/api/check_integration.go [hunk 2]
```diff
@@ -49,37 +49,47 @@ var (
...
+	return out
+}
+
+func checkAwsS3Integration(input *models.CheckIntegrationInput) *models.SourceIntegrationHealth {
+	out := &models.SourceIntegrationHealth{
+		AWSAccountID:    aws.StringValue(input.AWSAccountID),
+		IntegrationType: aws.StringValue(input.IntegrationType),
+	}
+	var roleCreds *credentials.Credentials
+	logProcessingRole := generateLogProcessingRoleArn(*input.AWSAccountID, *input.IntegrationLabel)
```
Comment (nhakmiller) [inline internal/core/source_api/api/check_integration.go:86]: Same question as above about a nil account ID

File: api/lambda/source/models/api.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ type PutIntegrationInput struct {
 
 // PutIntegrationSettings are all the settings for the new integration.
 type PutIntegrationSettings struct {
-	AWSAccountID       *string   `genericapi:"redact" json:"awsAccountId" validate:"required,len=12,numeric"`
+	AWSAccountID       *string   `genericapi:"redact" json:"awsAccountId" validate:"omitempty,len=12,numeric"`
```
[the remark this answers is not in the document]
Comment (kostaspap) [reply api/lambda/source/models/api.go:72]: There are some integrations that don't require this... 
It is similar to the scenario that we have where the S3Bucket is marked `omitempty`, however it should never be empty in an integration of type `aws-s3`. In both cases we expect that frontend sends the appropriate request.

[the remark this answers is not in the document]
Comment (kostaspap) [reply api/lambda/source/models/api.go]: I reverted back this change as it was not required. Was afraid of scope creep if I were to implement input validation for the API inputs... as it was not the purpose of my PR. I am planning to introduce some changes to the API anyway to address this specific issue but on a separate PR I am working on.

File: internal/core/source_api/api/check_integration.go (hunk 1 above)
Comment (kostaspap) [reply internal/core/source_api/api/check_integration.go:68]: Added back the "required" check for the AWS Account ID

File: internal/core/source_api/api/check_integration.go (hunk 2 above)
Comment (kostaspap) [reply internal/core/source_api/api/check_integration.go:86]: Added back the check in the API.

**Pull request merged**